libyul/AsmParser.h

```diff
@@ -35,6 +35,7 @@
 #include <memory>
 #include <variant>
 #include <vector>
+#include <string>
```

Comment:
The code you added does not even use `string`.
```suggestion
#include <string_view>
```